elastic/tests/test_elastic.py

```diff
@@ -0,0 +1,306 @@
+                                    status=ESCheck.OK,  # OK
+                                    tags=good_sc_tags + TAGS)
+
+    # Assert service metadata
+    # self.assertServiceMetadata(['version'], count=3)
+    # FIXME: 0.90.13 returns randomly a red status instead of yellow,
+    # so we don't do a coverage test for it
+    # Remove me when we stop supporting 0.90.x (not supported anymore by ES)
+    if get_es_version() != [0, 90, 13]:
+        # Warning because elasticsearch status should be yellow, according to
+        # http://chrissimpson.co.uk/elasticsearch-yellow-cluster-status-explained.html
+        aggregator.assert_service_check('elasticsearch.cluster_health')
+
+
+def test_config_parser(aggregator, spin_up_elastic):
+    elastic_check = ESCheck(CHECK_NAME, {}, {})
+    instance = {
+        "username": "user",
+        "password": "pass",
+        "is_external": "yes",
+        "url": "http://foo.bar",
+        "tags": ["a", "b:c"],
+    }
+    c = elastic_check.get_instance_config(instance)
+    assert c.username == "user"
+    assert c.password == "pass"
+    assert c.cluster_stats is True
+    assert c.url == "http://foo.bar"
+    assert c.tags == ["url:http://foo.bar", "a", "b:c"]
+    assert c.timeout == elastic_check.DEFAULT_TIMEOUT
+    assert c.service_check_tags == ["host:foo.bar", "port:None", "a", "b:c"]
+
+    instance = {
+        "url": "http://192.168.42.42:12999",
+        "timeout": 15}
+    c = elastic_check.get_instance_config(instance)
+    assert c.username is None
+    assert c.password is None
+    assert c.cluster_stats is False
+    assert c.url == "http://192.168.42.42:12999"
+    assert c.tags == ["url:http://192.168.42.42:12999"]
+    assert c.timeout == 15
+    assert c.service_check_tags == ["host:192.168.42.42", "port:12999"]
+
+    instance = {
+        "username": "user",
+        "password": "pass",
+        "url": "https://foo.bar:9200",
+        "ssl_verify": "true",
+        "ssl_cert": "/path/to/cert.pem",
+        "ssl_key": "/path/to/cert.key",
+    }
+    c = elastic_check.get_instance_config(instance)
+    assert c.username == "user"
+    assert c.password == "pass"
+    assert c.cluster_stats is False
+    assert c.url == "https://foo.bar:9200"
+    assert c.tags == ["url:https://foo.bar:9200"]
+    assert c.timeout == elastic_check.DEFAULT_TIMEOUT
+    assert c.service_check_tags == ["host:foo.bar", "port:9200"]
+    assert c.ssl_verify == "true"
+    assert c.ssl_cert == "/path/to/cert.pem"
+    assert c.ssl_key == "/path/to/cert.key"
+
+
+def test_pshard_metrics(aggregator, spin_up_elastic):
+    """ Tests that the pshard related metrics are forwarded and that the
+        document count for primary indexes is twice smaller as the global
+        document count when "number_of_replicas" is set to 1 """
+    elastic_latency = 10
+    config = {'url': 'http://localhost:9200', 'pshard_stats': True, 'username': USER, 'password': PASSWORD}
```

Comment:
Substitute every localhost with `HOST`.

Reply:
For the urls that mention `localhost`, I substituted with `URL` since it's composed with `HOST`